frontend/src/metabase/visualizations/components/settings/ChartSettingAddRemoveColumns/ChartSettingAddRemoveColumns.unit.spec.tsx

```diff
@@ -0,0 +1,221 @@
+import userEvent from "@testing-library/user-event";
+
+import { renderWithProviders, screen, within } from "__support__/ui";
+
+import { createMockTableColumnOrderSetting } from "metabase-types/api/mocks";
+import {
+  ORDERS,
+  ORDERS_ID,
+  PRODUCTS,
+  createOrdersTableDatasetColumns,
+} from "metabase-types/api/mocks/presets";
+
+import type {
+  FieldReference,
+  TableColumnOrderSetting,
+} from "metabase-types/api";
+
+import {
+  DEFAULT_QUERY,
+  SAMPLE_DATABASE,
+  SAMPLE_METADATA,
+} from "metabase-lib/test-helpers";
+import Question from "metabase-lib/Question";
+
+import { ChartSettingAddRemoveColumns } from "./ChartSettingAddRemoveColumns";
+
+const COLUMN_SETTINGS = [
+  createMockTableColumnOrderSetting({
+    name: "TOTAL",
+    fieldRef: ["field", ORDERS.TOTAL, null],
+    enabled: true,
+  }),
+  createMockTableColumnOrderSetting({
+    name: "ID",
+    fieldRef: ["field", ORDERS.ID, null],
+    enabled: true,
+  }),
+  createMockTableColumnOrderSetting({
+    name: "TAX",
+    fieldRef: ["field", ORDERS.TAX, null],
+    enabled: false,
+  }),
+  createMockTableColumnOrderSetting({
+    name: "SUBTOTAL",
+    fieldRef: ["field", ORDERS.SUBTOTAL, null],
+    enabled: false,
+  }),
+];
+
+const setup = ({
+  value = COLUMN_SETTINGS,
+  question = new Question(
+    {
+      dataset_query: DEFAULT_QUERY,
+    },
+    SAMPLE_METADATA,
+  ),
+} = {}) => {
+  const onChange = jest.fn();
+  const onWidgetOverride = jest.fn();
+
+  renderWithProviders(
+    <ChartSettingAddRemoveColumns
+      value={value}
+      question={question}
+      onChange={onChange}
+      onWidgetOverride={onWidgetOverride}
+    />,
+  );
+
+  return {
+    onChange,
+  };
+};
+
+describe("AddRemoveColumns", () => {
```

Comment:
Can we add a bit more specs on different corner cases here? Historically, this functionality was  a bit fragile so having more specs would be a huge quality improvement

Reply:
I can try to add more here, but many corner cases on things like question structure are handled in `table-column-settings.cy.spec.js`